test/specs/column/tests/01.columns_list.js

```diff
@@ -757,6 +760,88 @@ exports.execute = function (options) {
                 areSameColumnList(compactRef.generateColumnsList(), compactColumns);
                 areSameColumnList(entryEditRef.generateColumnsList(), entryEditRef.columns);
             });
+
+            describe('with system columns heuristic config options defined,', function () {
+                function systemColumnsHeuristicsMode(context) {
+                    var mode = null;
+                    if (context == 'compact') {
+                        mode = true;
+                    } else if (context == 'detailed') {
+                        mode = ['RMB', 'RCT'];
+                    }
+
+                    return mode;
+                }
+
+                var compactSystemColumnsModeRef, compactBriefSystemColumnsModeRef, compactBriefInlineSystemColumnsModeRef,
+                    compactSelectSystemColumnsModeRef, compactSystemColumnsModeColumns, compactBriefSystemColumnsModeColumns,
+                    compactBriefInlineSystemColumnsModeColumns, compactSelectSystemColumnsModeColumns;
+
+                var detailedSystemColumnsModeRef, detailedSystemColumnsModeColumns;
+                beforeAll(function (done) {
+                    // set this here so it doesn't affect above columns list tests
+                    options.ermRest.systemColumnsHeuristicsMode(systemColumnsHeuristicsMode);
+                    options.ermRest.resolve(singleEnitityUriSystemColumnsHeuristics, {
+                        cid: "test"
+                    }).then(function (response) {
+                        compactSystemColumnsModeRef = response.contextualize.compact;
```

Comment:
I don't think you need to test all sub-contexts of `compact` here. As I mentioned I think it's chaise's responsibility to consume the contexts correctly.